analysisservice/src/main/java/nu/yona/server/analysis/service/AnalysisEngineService.java

```diff
@@ -359,33 +405,29 @@ private void updateActivityEndTime(UserAnonymizedEntityHolder userAnonymizedHold
 		userAnonymizedHolder.getEntity();
 	}
 
-	private boolean shouldUpdateCache(ActivityDto lastRegisteredActivity, Activity newOrUpdatedActivity)
+	private boolean shouldUpdateCache(Optional<ActivityDto> lastRegisteredActivity, Activity newOrUpdatedActivity)
 	{
-		if (lastRegisteredActivity == null)
+		if (!lastRegisteredActivity.isPresent())
 		{
 			return true;
 		}
 
 		// do not update the cache if the new or updated activity occurs earlier than the last registered activity
 		return !newOrUpdatedActivity.getEndTime().atZone(newOrUpdatedActivity.getTimeZone())
```

Comment:
You could use `getEndTimeAsZonedDateTime` here for simplicity

Reply:
Good one!